clients/src/main/java/org/apache/kafka/clients/admin/DescribeReplicaLogDirResult.java

```diff
@@ -0,0 +1,132 @@
+        return futures;
+    }
+
+    /**
+     * Return a future which succeeds if log directory information of all replicas are available
+     */
+    public KafkaFuture<Map<TopicPartitionReplica, ReplicaLogDirInfo>> all() {
+        return KafkaFuture.allOf(futures.values().toArray(new KafkaFuture[0])).
+            thenApply(new KafkaFuture.Function<Void, Map<TopicPartitionReplica, ReplicaLogDirInfo>>() {
+                @Override
+                public Map<TopicPartitionReplica, ReplicaLogDirInfo> apply(Void v) {
+                    Map<TopicPartitionReplica, ReplicaLogDirInfo> replicaLogDirInfos = new HashMap<>();
+                    for (Map.Entry<TopicPartitionReplica, KafkaFuture<ReplicaLogDirInfo>> entry : futures.entrySet()) {
+                        try {
+                            replicaLogDirInfos.put(entry.getKey(), entry.getValue().get());
+                        } catch (InterruptedException | ExecutionException e) {
+                            // This should be unreachable, because allOf ensured that all the futures completed successfully.
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    return replicaLogDirInfos;
+                }
+            });
+    }
+
+    static public class ReplicaLogDirInfo {
+        // The log directory of the primary replica of this partition on the given broker.
+        // Null if the primary replica is not found for this partition on the given broker.
+        private final String currentReplicaLogDir;
+        // Defined as max(HW of partition - LEO of the primary replica, 0).
+        private final long currentReplicaOffsetLag;
+        // The log directory of the temporary replica of this partition on the given broker.
+        // Null if the temporary replica is not found for this partition on the given broker.
+        private final String temporaryReplicaLogDir;
+        // The LEO of the primary replica - LEO of the temporary replica.
+        // -1 if either primary or the temporary replica is not found for this partition on the given broker.
+        private final long temporaryReplicaOffsetLag;
+
+        ReplicaLogDirInfo() {
+            this(null, DescribeLogDirsResponse.INVALID_OFFSET_LAG, null, DescribeLogDirsResponse.INVALID_OFFSET_LAG);
+        }
+
+        ReplicaLogDirInfo(String currentReplicaLogDir,
+                                 long currentReplicaOffsetLag,
+                                 String temporaryReplicaLogDir,
+                                 long temporaryReplicaOffsetLag) {
+            this.currentReplicaLogDir = currentReplicaLogDir;
+            this.currentReplicaOffsetLag = currentReplicaOffsetLag;
+            this.temporaryReplicaLogDir = temporaryReplicaLogDir;
+            this.temporaryReplicaOffsetLag = temporaryReplicaOffsetLag;
+        }
+
+        public String getCurrentReplicaLogDir() {
```

Comment:
getPermanentReplicaLogDir seems to match getTemporaryReplicaLogDir better. Ditto on getCurrentReplicaOffsetLag.

Reply:
I am bit concerned that `getPermanentReplicaLogDir` may mislead user into thinking log directory of this partition will never change. How about we rename `getTemporaryReplicaLogDir` to `getFutureReplicaLogDir`?

Reply:
Perhaps use getCurrentReplicaLogDir and get getNewReplicaLogDir?

Reply:
Sure. I will have updated the patch to use `getNewReplicaLogDir` and rename other fields as appropriate.

Reply:
@junrao After discussing with @becketqin, I think it may be better to use `getFutureReplicaLogDir`.

If we were to use `getNewReplicaLogDir`, do you think we should rename `is_temporary` field in `DESCRIBE_LOG_DIRS_RESPONSE_V0` to `is_new`? If so, it kinds of collide with the `is_new` field in `LEADER_AND_ISR_REQUEST_PARTITION_STATE_V1`. It seems a bit confusing.

Also, if we were to use `getFutureLogDir`, are you OK with renaming the field `is_temporary` to `is_future` in `DESCRIBE_LOG_DIRS_RESPONSE_V0`?

Reply:
Ok, future is fine then. Changing is_temporary to is_future also sounds good.